rackcity/models/asset.py

```diff
@@ -77,3 +78,28 @@ def save(self, *args, **kwargs):
                         self.asset_number = asset_number
                         break
             super(Asset, self).save(*args, **kwargs)
+            self.add_network_ports()
+            self.add_power_ports()
+
+    def add_network_ports(self):
+        from rackcity.models import NetworkPort
+        if len(NetworkPort.objects.filter(asset=self.id)) == 0:  # only add ports for new assets
+            model = self.model
+            for network_port_name in model.network_ports:
+                network_port = NetworkPort(
+                    asset=self,
+                    port_name=network_port_name,
+                )
+                network_port.save()
+
+    def add_power_ports(self):
+        from rackcity.models import PowerPort
```

Comment:
Was getting funky stuff with circular imports because Asset imports *Port and *Port imports Asset.  Looked up solutions and the only one that wasn't crazy is importing it in the method.  I think I could also call it by its fully qualified name instead of importing but not sure that's any better

Reply:
stars, they're just like us - everyone has some circular dependencies in their closet